clients/runtime/Cargo.toml

```diff
@@ -14,9 +14,9 @@ testing-utils = [
     "tempdir",
     "rand",
     "testchain",
-    "testchain-runtime",
+    "testchain-runtime/testing-utils",
```

Comment:
addresses https://github.com/pendulum-chain/spacewalk/blob/bda384bcc6db3da88198c714851a261dcdea1063/testchain/runtime/src/lib.rs#L451-L460.

this way, testing ONE intergration test case will work.

Reply:
What do you mean with 'testing ONE integration test case will work'? Do these 'testing-utils' import solve the problem you linked to?

Reply:
when we wanted to test only 1 case: 
```
cargo test test_redeem_succeeds_on_mainnet -p vault --features standalone-metadata,integration
```

It goes inside the 
```rust
if #[cfg(not(feature = "testing-utils"))] {
```
block of code.

For 
```
cargo test --all --all-features
```
it  doesn't. 

My theory is: when running a _singular test case_ in the **`vault`**, it doesn't make use of the `runtime`'s _`dev-dependencies`_. But the `runtime` clearly specifies to use the _`testing-utils`_ feature of **`testchain-runtime`**.
https://github.com/pendulum-chain/spacewalk/blob/bda384bcc6db3da88198c714851a261dcdea1063/clients/runtime/Cargo.toml#L71


The `vault`'s `dev-dependencies` calls for the _`testing-utils`_ feature of the `runtime`. But as you can see in `runtime`, its _`testing-utils`_ feature is using the defaults of `testchain-runtime` and `oracle`.
That's why it got confusing why the `cfg-if` didn't work in `testchain-runtime`.